### PR TITLE
refactored service to get all apis in a category

### PR DESCRIPTION
**Pull request by alahirajeffrey**
fix for issue https://github.com/Zummit-Africa-Inc/zapi-lambda-core/issues/47

### CODE REVIEW

File: src/categories/categories.service.ts [hunk 1]
```diff
@@ -57,10 +57,12 @@ export class CategoriesService {
   async getAllApis(categoryId: string){
     try {
       //check if category exists
-      const categoryExists = await this.categoryRepo.findOne({where:{id:categoryId}})
+      const category = await this.categoryRepo.findOne({
+        where:{id:categoryId},
+        relations:{api: true}})
```
Comment (PhilemonBrain) [inline src/categories/categories.service.ts:62]: please jefferey why did u add the relations property?

Comment (PhilemonBrain) [reply src/categories/categories.service.ts:62]: @alahirajeffrey

Comment (alahirajeffrey) [reply src/categories/categories.service.ts]: @PhilemonBrain I made the changes we spoke about on the call and pushed it. I then decided to test it out again and its not working again

Comment (alahirajeffrey) [reply src/categories/categories.service.ts]: @Brain I think the problem was that I switched branches and tried testing again.

File: src/entities/category.entity.ts [hunk 2]
```diff
@@ -13,5 +13,6 @@ export class Category extends SharedEntity {
   description: string;
 
   @OneToMany(() => Api, (api) => api.category)
+  @JoinColumn()
```
Comment (johnayinde) [inline src/entities/category.entity.ts:16]: You can remove the JoinColumn on that line  it's not necessary

**Review by johnayinde — CHANGES_REQUESTED** [review]

**Review by PhilemonBrain — APPROVED** [review]

**Pull request merged**